NuGet.Docs/Consume/Credential-Providers.md

```diff
@@ -21,12 +22,87 @@ multiple files/assemblies, using subfolders can help keep the Providers organize
 Credential Providers can also be stored anywhere and then made accessible to NuGet.exe via an environment variable. To use a Credential Provider this way, set the `%NUGET_CREDENTIALPROVIDERS_PATH%` to the location of your Provider. The variable can be a semicolon-separated list, e.g. 
 `path1;path2`, if you have have multiple Credential Providers in different locations.
 
-## Using a credential provider alongside NuGet.exe
+## Using a Credential Provider Alongside NuGet.exe
 
 Credential Providers can also be placed in the same folder as NuGet.exe.
 
+## Using Multiple Credential Providers
+
+The NuGet.exe client will search the above locations, in order, for Credential Providers.
+Specifically, it will search for any file that matches the pattern `CredentialProvider*.exe`
+and load each provider in the order it's found. If two Credential Providers are found in the
+same directory, they will be loaded in alphabetical order.
+
 ## Available Credential Providers
 
 Some of the available Credential Providers are:
 
-* [Visual Studio Online Credential Provider](https://www.visualstudio.com/get-started/package/use/auth.md#vsts-credential-provider)
\ No newline at end of file
+* [Visual Studio Team Services Credential Provider](https://www.visualstudio.com/get-started/package/use/auth.md#vsts-credential-provider)
+
+# Creating a Credential Provider
+
+The NuGet 3.3 client implements an internal CredentialService that is used to acquire credentials.
+The CredentialService has a list of built-in and plug-in Credential Providers.
+Each provider is tried sequentially until credentials are acquired.
+
+Built-in providers, such as the existing providers to fetch credentials from nuget.config and to
+prompt the user on the command line, execute in-process. 
+
+Two Credential Provider extensibility points are available:
+
+1. An internal-only extensibility point that allows users compiling their own client to inject
```

Comment:
We shouldn't document an internal extensibility point -- not to be secret, but it is not helpful to learn about an extensibility point you can't use.